### PR TITLE
Upgrade dev dependencies to stable release if possible and lock others.

### LINKED ISSUE #2474: Epic: Prepare platform for upgrade to Drupal 9
# Motivation

To stay current with the most recent version of Drupal

# Proposed solution

The following updates will allow us to update to Drupal 9.0.x. Some issues may require checking the warnings generated in the Upgrade Status report available on a site when viewing it locally.

- [x] uiowa/uiowa_bar#59: Update uiowa_bar module to be D9 compatible.
- [x] ~~#2592 Bump `drupal/big_pipe_sessionless:^2.0`~~ - This has to be done at the same time as moving to D9
- [x] #2593 Bump `drupal/components:^2.0@beta`
- [x] ~~#2594 Apply patch from https://www.drupal.org/project/config_ignore/issues/3150462 to make `drupal/config_ignore` compatible.~~ - This is no longer necessary because of #2772 
- [x] #2455 Bump `drupal/devel:^4.0`
- [x] ~~Resolve issues with drupal/field_group, possibly by applying the patch here: https://www.drupal.org/project/field_group/issues/3168213~~ - Not necessary, the library will get updated via D9 upgrade.
- [x] #2872 Bump `composer update drupal/fragments`
- [x] ~~Determine whether additional action is necessary to resolve `drupal/google_analytics` warnings in Upgrade Status~~ - Not necessary as all the warnings relate to an extension being uninstalled that is only required for tests.
- [x] Bump `drupal/honeypot:^2.0`
- [x] #2595 Bump `drupal/imagecache_external:^3.0.`
- [x] Determine whether additional action is necessary to resolve `drupal/media_library_theme_reset` warnings in Upgrade Status
- [x] Resolve `drupal/menu_block` compatibility issues - needs more research about whether patches are compatible with stable version or whether current dev commit hash we are pinned to works
- [x] Resolve `drupal/metatag` compatibility issues - needs more research about whether patch is compatible with stable version or we can roll with the dev version
- [x] ~~Determine whether additional action is necessary to resolve `drupal/paragraphs` warnings in Upgrade Status~~ - Not necessary as all warnings are related to tests or modules that are not installed.
- [x] Investigate purge warnings in Upgrade Status
- [x] ~~Determine whether additional action is necessary to resolve `drupal/search_api` warnings in Upgrade Status~~ - Not necessary as all warnings are related to tests.
- [x] Determine whether additional action is necessary to resolve `drupal/simple_sitemap` warning in Upgrade Status
- [x] #2596 Bump `drupal/webform:^6.0`

### LINKED ISSUE #2593: Update Component Libraries to 8.x-2.0-beta3
# Motivation

In preparation for moving to Drupal 9, updating to a compatible version of Component Libraries is necessary.

# Proposed solution

- [x] `composer require 'drupal/components:^2.0@beta'`
- [x] Test that there are no errors when UIDS component libraries are referenced.

### LINKED ISSUE #2793: Update or pin dev modules to commit hashes to prevent unexpected updates
In trying to add a new patch to `drupal/core` in a PR, I kept running into an issue where several modules, including Metatag, were also taking the opportunity to update themselves. Essentially, if we have `-dev` modules, anytime we run `composer update --lock`, this can potentially happen. In the case of Metatag, this was the diff between the two versions: https://git.drupalcode.org/project/metatag/-/compare/c32d6f8...e46ef6b

- [x] "drupal/layout_builder_shortcuts": "1.0.x-dev"
- [x] "drupal/lb_direct_add": "1.x-dev"
- [x] "drupal/media_entity_facebook": "2.x-dev@dev"
- [x] "drupal/media_entity_file_replace": "1.x-dev"
- [x] "drupal/media_entity_instagram": "3.x-dev"
- [x] "drupal/menu_block": "1.x-dev#fbfb8fe"
- [x] "drupal/metatag": "1.x-dev"
- [x] "drupal/redirect": "1.x-dev"
- [x] "kartsims/easysvg": "dev-master"
- [x] "uiowa/block_content_template": "dev-master"

### LINKED ISSUE #2910: Update Readme with new core update steps
https://github.com/uiowa/uiowa#updating-dependencies with the core-recommended update steps?

### PR DESCRIPTION
**Pull request by richardbporter**
Resolves #2793 
Resolves #2593
Resolves #2910
Finally resolves #2474 

### How to test
- Review https://github.com/uiowa/uiowa/pull/2912/files
  - Review the updates to the readme about updating drupal/core and contrib packages.
    - This includes removal of `drupal/core` which resulted in some updates: https://github.com/uiowa/uiowa/pull/2912/commits/9c38c9ecf20e4a696409aeb776b76e4294a1947f
  - Review the readme section on locked dependencies.
  - Review the changes in the composer.json file.
- Pull and run `composer install`.
  - Check the Composer output for anything unusual...
  - Note other updates that might need additional testing and outline those here.
- Test out the menu_block update.
  - Sync sites with block config which includes menu_block settings: default, hr.uiowa.edu
    -  **Need help** here with testing criteria.
  - Some directories have block config without menu_block settings which were not affected: sitenow.uiowa.edu, policy.clas.uiowa.edu, uids_base/config/optional, 
- TBD.

### CODE REVIEW

File: config/README.md [hunk 1]
```diff
@@ -48,14 +48,10 @@ See config/features/README.md for more information.
...
 
 **DO NOT** share theme-dependent configuration in the sync directory if your
-site split has blacklisted a custom theme. The shared configuration will be
+site split has complete-split a custom theme. The shared configuration will be
 deleted on export which will break BLT's configuration integrity checks. An
 example of this would be a block.
-
-**DO NOT** graylist configuration that depends on blacklisted configuration
-unless you intend to discard changes to the sync directory after every export.
-See https://www.drupal.org/project/config_split/issues/3117841 for more information.
```
Comment (richardbporter) [inline config/README.md:61]: This issue looks to be committed in the release we are using.

**Comment by joewhitsitt** [comment]
Would you be willing to add https://github.com/uiowa/uiowa/pull/2900 to this PR so we can do less composer.lock stuff?

**Comment by richardbporter** [comment]
> Would you be willing to add #2900 to this PR so we can do less composer.lock stuff?

SUre

**Comment by pyrello** [comment]
I don't know if it is reasonable to also look through this and see if any of these other changes can be done at the same time? #2474

**Comment by richardbporter** [comment]
Might as well!

**Comment by richardbporter** [comment]
> Might as well!

This is not sarcasm. :)

**Comment by richardbporter** [comment]
I think is ready for an initial review but I am looking for help outlining test criteria.

File: README.md [hunk 2]
```diff
@@ -98,20 +98,33 @@ $settings['config_sync_directory'] = DRUPAL_ROOT . '/config/' . $site_dir;
...
+## Contrib
+You can run `composer update package/name` to update additional dependencies. The output from the Composer commands can be used as the long text for commit messages. Ideally, each package update would be one commit for clarity and easier reverting.
+
+### Locked Packages
+The packages below are locked at specific SHAs and will not update using the method described above. They should be periodically checked for new stable releases and updated, if viable.
+
+| Package                               | Reason                   |
+| ------------------------------------- | ------------------------ |
+| drupal/layout_builder_shortcuts       | No stable release.       |
+| drupal/lb_direct_add                  | No stable release.       |
```
Comment (pyrello) [inline README.md:117]: Putting the pressure on ;)

**Comment by richardbporter** [comment]
I forgot to mention I don't understand this change which I exported after a fresh install: https://github.com/uiowa/uiowa/pull/2912/commits/0c86c9eb8262cda07ec5073c92163763c937f7ea

**Comment by richardbporter** [comment]
Yea it just seems odd because I thought that was menu_block config so I'm not sure why its on config/default/core.entity_view_display.node.page.default.yml.

**Comment by richardbporter** [comment]
I'm deploying this to dev now.

**Comment by pyrello** [comment]
> Yea it just seems odd because I thought that was menu_block config so I'm not sure why its on `config/default/core.entity_view_display.node.page.default.yml`.

This makes sense because the menu block is embedded in LB for the page template and this how that gets represented.

**Comment by joewhitsitt** [comment]
pulling down to test

~~need to do more testing, but I think this breaks our fb/instagram stuff~~

you need to have the secret and app id from facebook to test embeds locally. instagram seems to be broken in prod so technically unrelated to this PR... no big thang just an endpoint issue: https://www.drupal.org/project/media_entity_instagram/issues/3190830

**Comment by richardbporter** [comment]
I think all the media_* module updates didn't result in any changes. We were using a hash that must've got tagged as the latest release.

**Review by joewhitsitt — APPROVED** [review]
Nothing seems to be broken that wasn't previously broken.

**Comment by joewhitsitt** [comment]
Once this deploys, I can delete this https://github.com/joewhitsitt/easysvg right @richardbporter?

**Comment by richardbporter** [comment]
Might be good to merge after the next release is tagged.

**Comment by pyrello** [comment]
"A gentleman who goes by the name Upgrayedd..."

**Comment by richardbporter** [comment]
> "A gentleman who goes by the name Upgrayedd..."

YES

**Pull request merged**